powerplantmatching/osm/quality/coverage.py

```diff
@@ -0,0 +1,881 @@
+                if country_code not in cached_countries:
+                    continue
+
+                country_name = cached_countries[country_code]["name"]
+
+                try:
+                    live_counts = client.count_country_elements(country_name, "both")
+
+                    cached_countries[country_code]["plants_live"] = live_counts.get(
+                        "plants", -1
+                    )
+                    cached_countries[country_code]["generators_live"] = live_counts.get(
+                        "generators", -1
+                    )
+
+                    if (
+                        live_counts.get("plants", -1) >= 0
+                        and live_counts.get("generators", -1) >= 0
+                    ):
+                        cached_countries[country_code]["total_live"] = (
+                            live_counts["plants"] + live_counts["generators"]
+                        )
+
+                        cached_countries[country_code]["plants_diff"] = (
+                            cached_countries[country_code]["plants_cached"]
+                            - live_counts["plants"]
+                        )
+                        cached_countries[country_code]["generators_diff"] = (
+                            cached_countries[country_code]["generators_cached"]
+                            - live_counts["generators"]
+                        )
+
+                        plants_ratio = (
+                            cached_countries[country_code]["plants_cached"]
+                            / live_counts["plants"]
+                            if live_counts["plants"] > 0
+                            else 1.0
+                        )
+                        generators_ratio = (
+                            cached_countries[country_code]["generators_cached"]
+                            / live_counts["generators"]
+                            if live_counts["generators"] > 0
+                            else 1.0
+                        )
+
+                        if (
+                            plants_ratio < outdated_threshold
+                            or generators_ratio < outdated_threshold
+                        ):
+                            cached_countries[country_code]["cache_status"] = "outdated"
+                        elif plants_ratio > 1.0 or generators_ratio > 1.0:
+                            cached_countries[country_code]["cache_status"] = "ahead"
+                        else:
+                            cached_countries[country_code]["cache_status"] = "current"
+                    else:
+                        cached_countries[country_code]["cache_status"] = "error"
+
+                except Exception as e:
+                    logger.error(
+                        f"Error checking live counts for {country_name}: {str(e)}"
+                    )
+                    cached_countries[country_code]["cache_status"] = "error"
+
+    if return_data:
```

Comment:
instead of spending another 100 lines of code in this function, I would rather create a function which works on the output dict and just handles the printing.
That way, we can spare the `return_data` parameter as well and receive cleaner code.